core/src/main/java/com/linecorp/armeria/server/cors/CorsDecoratorsFactoryFunction.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.cors;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.annotation.DecoratorFactoryFunction;
+import com.linecorp.armeria.server.annotation.decorator.CorsDecorator;
+import com.linecorp.armeria.server.annotation.decorator.CorsDecorators;
+
+/**
+ * A factory which creates a {@link CorsService} decorator when more than two {@link CorsDecorator}s are added.
+ */
+public final class CorsDecoratorsFactoryFunction implements DecoratorFactoryFunction<CorsDecorators> {
+
+    @Override
+    public Function<Service<HttpRequest, HttpResponse>,
+            ? extends Service<HttpRequest, HttpResponse>> newDecorator(CorsDecorators parameter) {
+        ensureValidConfig(parameter);
+        final CorsDecorator[] policies = parameter.value();
+        final CorsDecorator corsDecorator = policies[0];
+        final CorsServiceBuilder cb = CorsServiceBuilder.forOrigins(corsDecorator.origins());
+        if (parameter.shortCircuit()) {
+            cb.shortCircuit();
+        }
+        cb.firstPolicyBuilder.setConfig(corsDecorator);
+        for (int i = 1; i < policies.length; i++) {
+            final CorsPolicyBuilder builder = new CorsPolicyBuilder(policies[i].origins());
+            builder.setConfig(policies[i]);
+            cb.addPolicy(builder.build());
+        }
+        return cb.newDecorator();
+    }
+
+    private static void ensureValidConfig(CorsDecorators conf) {
+        checkState(conf.value().length > 0, "value() should not be empty.");
+        final boolean anyOrigin = Arrays.stream(conf.value()).anyMatch(
+                c -> Arrays.asList(c.origins()).contains("*"));
+        final CorsDecorator[] policies = conf.value();
```

Comment:
We can move this line up to the first line of this method and reuse the parameter `policies`.